### PR TITLE
[timelines] allow loading of pes by id.

### PR DESCRIPTION
**Pull request by nkenneweg**
Implements [2470](https://www.openproject.org/issues/2470)

### CODE REVIEW

**Comment by hschink** [comment]
Travis found a failing spec: `rspec ./spec/controllers/api/v2/planning_elements_controller_spec.rb:112`

**Comment by nkenneweg** [comment]
I added some specs and fixed the bug. Will run this on ci and then it should be ready to merge!

File: spec/controllers/api/v2/planning_elements_controller_spec.rb [hunk 1]
```diff
@@ -108,106 +108,202 @@ def self.become_member_with_delete_planning_element_permissions(&block)
...
+          end          
+        end
+      end
+    end
 
+    describe 'w/ list of projects' do
+      describe 'w/ an unknown project' do
         it 'renders a 404 Not Found page' do
-          get 'index', :project_id => 'project_x,project_b', :format => 'xml'
+          get 'index', :project_id => '4711', :format => 'xml'
```
Comment (ulferts) [inline spec/controllers/api/v2/planning_elements_controller_spec.rb]: this no longer is a list of projects

File: spec/controllers/api/v2/planning_elements_controller_spec.rb [hunk 2]
```diff
@@ -219,26 +315,42 @@ def self.become_member_with_delete_planning_element_permissions(&block)
...
+                FactoryGirl.create(:work_package, :project_id => project_b.id),
+                FactoryGirl.create(:work_package, :project_id => project_b.id)
+              ]
+              # adding another planning element, just to make sure, that the
+              # result set is properly filtered
+              FactoryGirl.create(:work_package, :project_id => project_c.id)
+            end
+
+            it 'assigns a planning_elements array containing all three elements' do
+              get 'index', :project_id => 'project_a,project_b', :format => 'xml'
```
Comment (hschink) [inline spec/controllers/api/v2/planning_elements_controller_spec.rb]: IMHO, `get 'index', :project_id => 'project_a,project_b', :format => 'xml'` is part of the precondition for the spec. Thus, you should move this to the end of the `before` block. Then, you don't need to replicate this code in every `it`.

You may check other locations for the same pattern. :eyes:

Comment (nkenneweg) [inline spec/controllers/api/v2/planning_elements_controller_spec.rb]: I did not write that code, but I think your request is valid.

**Pull request merged**